src/Symfony/Component/HttpKernel/EventListener/AbstractTestSessionListener.php

```diff
@@ -45,6 +47,7 @@ public function onKernelRequest(GetResponseEvent $event)
 
         if ($cookies->has($session->getName())) {
             $session->setId($cookies->get($session->getName()));
```

Comment:
I'd suggest to move this line below the current next one, and write it as:
`$session->setId($this->initialSessionId)`